test/memory-usage/tail-big-log-file.js

```diff
@@ -0,0 +1,73 @@
+'use strict'
+
+const fs = require('fs')
+const os = require('os')
+const path = require('path')
+const readline = require('readline')
+const {spawn} = require('child_process')
+const {test} = require('tap')
+const TailFile = require('../../index.js')
+
+const ALL_DONE_AFTER_TIME = 5000 // Close tail after 5 seconds of inactivity
+const shellCmd = path.join(__dirname, 'populate-big-log.sh')
+
+test('Memory consumption', async (t) => {
+  const LOG_FILE_NAME = 'big-log-file.log'
+  const testDir = os.tmpdir()
+  const LOG_FILE_PATH = path.join(testDir, LOG_FILE_NAME)
+
+  t.test('Create a stub for the large log file', async (t) => {
+    await fs.promises.writeFile(LOG_FILE_PATH, '', 'utf8')
+  })
+
+  t.test('Report memory consumptions when lots of large lines are consumed', (t) => {
+    const tail = new TailFile(LOG_FILE_PATH)
+      .on('error', (err) => {
+        t.fail(err)
+      })
+
+    t.teardown(async () => {
+      await tail.quit()
+    })
+
+    const lineSplitter = readline.createInterface({
+      input: tail
+    })
+
+    let counter = 0
+    let timer = null
+    // Resident Set Size - occupied memory including heap, segment and stack
+    let rssMin = Infinity
+    let rssMax = 0
+
+    lineSplitter.on('line', (line) => {
+      clearTimeout(timer)
+      if (++counter % 10000 === 0) {
+        const usage = process.memoryUsage()
+        rssMin = Math.min(usage.rss, rssMin)
+        rssMax = Math.max(usage.rss, rssMax)
```

Comment:
if memory usage is consistent(ish) across runs - and in theory it should be - we could assert on the maximum being within range. it would be cool if we could fail the build if the changeset balloons memory usage.

not feedback here, just a thought

Reply:
Yeah, that'd be nice to make it more functional, but the original intention was just as an FYI.  It does vary slightly across executions, so creating an upper limit is not out of the question if we do this in the future.